src/readii_2_roqc/utils/settings.py

```
@@ -0,0 +1,168 @@
+    _regions, _permutations, crop, resize = get_readii_settings(dataset_config)
+
+    try:
+        # Path to find existing readii index output for checking existing outputs
+        if crop != "" and resize != []:
+            extract_index_filepath = extract_features_dir.glob(f"{crop}_{get_resize_string(resize)}/{extract_method}_{dataset_name}_index.csv").__next__()
+        else:
+            extract_index_filepath = extract_features_dir.glob(f"original_*/{extract_method}_{dataset_name}_index.csv").__next__()
+    except StopIteration:
+        message = f"No {extract_method} index file was found for the specified settings"
+        logger.warning(message)
+        raise FileNotFoundError(message)
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)